src/Components/portfolio/mainPage.js

```diff
@@ -2,50 +2,107 @@ import React, { useEffect, useState } from "react";
 import PopCoinForm from "./popForm";
 import axios from "axios";
 
-import { Container, Header, Button, BtnDiv } from "./index.style";
+import {
+  Container,
+  Header,
+  Button,
+  BtnDiv,
+  Main,
+  ImageSection,
+  DataContainer,
+  MainContainer,
+  AssetDiv,
+  DeleteIcon,
+} from "./index.style";
+import { CoinMarketValue, UserCoinMarketValue } from "./list";
+
+const MarketCampare = ({ data, ...props }) => {
+  return (
+    <Main>
+      <MainContainer>
+        <ImageSection>
+          <DeleteIcon onClick={() => props.handleRemove(data.id)} />
+
+          <img src={data.image} alt="browser doesn't support" />
+          <p>{data.name}</p>
+        </ImageSection>
+        <DataContainer>
+          <CoinMarketValue data={data} />
+          <UserCoinMarketValue data={data} />
+        </DataContainer>
+      </MainContainer>
+    </Main>
+  );
+};
 
 const getLocalItems = () => {
   let list = localStorage.getItem("list");
-  console.log(list);
   if (list) {
     return JSON.parse(localStorage.getItem("list"));
   } else {
     return [];
   }
 };
+const GetCoinLocalStorage = () => {
+  let coins = localStorage.getItem("coin");
+  if (coins) {
+    return JSON.parse(localStorage.getItem("coin"));
+  } else {
+    return [];
+  }
+};
 
 function MainPage() {
   const [IsActive, setIsActive] = useState(false);
   const [list, setList] = useState(getLocalItems());
   const [coinList, setCoinList] = useState([]);
+  const [coinDataHouse, setCoinDataHouse] = useState(GetCoinLocalStorage());
   const [isError, setIsError] = useState(false);
 
-  const getCoinData = async (props) => {
+  useEffect(() => {
+    getCoinData();
+  }, []);
+
+  const getCoinData = async () => {
     try {
       const response = await axios(
-        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=150&page=1&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
+        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=300&page=1&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
       );
       setCoinList(response.data);
     } catch (err) {
       setIsError(true);
     }
   };
 
-  useEffect(() => {
-    getCoinData();
-  }, []);
+  const handleRemove = (item) => {
+    console.log("clicked item here", item);
+    const filterList = coinDataHouse.filter((element) => element.id !== item);
+    setCoinDataHouse(filterList);
+  };
 
   const handleClick = () => {
     setIsActive(!IsActive);
   };
-  const handleList = (item) => {
-    setList([...Object.values(list), item]);
+  const handleList = (items) => {
+    if (items) {
+      let str = items.value.toString().toLowerCase();
+      coinList.map((item) => {
+        if (item.id === str) {
+          const newItem = { ...item, items };
+          setCoinDataHouse([...coinDataHouse, newItem]);
+        }
+      });
```

Comment:
are you sure this is the best way to do this?

Reply:
Hey bro, what is the best approach I can do here?